components/dashboard/src/data/workspaces/create-workspace-mutation.ts

```diff
@@ -7,11 +7,38 @@
 import { GitpodServer } from "@gitpod/gitpod-protocol";
 import { useMutation } from "@tanstack/react-query";
 import { getGitpodService } from "../../service/service";
+import { useState } from "react";
 
 export const useCreateWorkspaceMutation = () => {
-    return useMutation({
+    const [isStarting, setIsStarting] = useState(false);
+    const mutation = useMutation({
         mutationFn: async (options: GitpodServer.CreateWorkspaceOptions) => {
             return await getGitpodService().server.createWorkspace(options);
         },
+        onMutate: async (options: GitpodServer.CreateWorkspaceOptions) => {
+            setIsStarting(true);
+        },
+        onError: (error) => {
+            console.error(error);
+            setIsStarting(false);
+        },
+        onSuccess: (result) => {
+            if (result && result.createdWorkspaceId) {
+                // successfully started a workspace, wait a bit before we allow to start another one
+                setTimeout(() => {
+                    setIsStarting(false);
+                }, 4000);
+            } else {
+                setIsStarting(false);
```

Comment:
Should this be `true`? As in, we haven't yet received the createdWorkspaceId, so we're still starting, no?

Reply:
`createWorkspace` either returns with workspace or with some error condition. When it returns with some non-workspace response we immediately set starting to false.